update PR template with def of shared backend

geoffcline opened the pull request:
based on the discussion in #58 I propose revising the pull request template.

This version clarifies when a group of domains is necessary.

<!-- Thanks for contributing! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] The PR isn't documenting something that would be a common practice among password managers (e.g. minimal length of 6)
- [x] The top-level JSON objects are sorted alphabetically 
- [x] There is no Well-Known URL for Changing Passwords (`https://example.com/.well-known/change-password`)
- [x] There are no [open pull requests](https://github.com/apple/password-manager-resources/pulls) for the same update.

igor-makarov commented:
@geoffcline perhaps the line should also give an example of what doesn't qualify? E.g. `google.co.il` redirecting to `accounts.google.com`.

Additionally, could you also split out the 'well-known' checklist item to a subheading titled "for password change URLs"?

geoffcline commented:
In my humble opinion, anything beyond a short example should be placed on a reference page. Perhaps I could also change the readme to have a section with this info, and link to it from the PR template?

I agree with splitting out the well known item.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Igor Makarov <notifications@github.com>
Sent: Sunday, June 7, 2020 12:52:08 AM
To: apple/password-manager-resources <password-manager-resources@noreply.github.com>
Cc: Geoffrey Cline <geoffreyc@outlook.com>; Mention <mention@noreply.github.com>
Subject: Re: [apple/password-manager-resources] update PR template with def of shared backend (#129)


@geoffcline<https://github.com/geoffcline> perhaps the line should also give an example of what doesn't qualify? E.g. google.co.il redirecting to accounts.google.com.

Additionally, could you also split out the 'well-known' checklist item to a subheading titled "for password change URLs"?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/apple/password-manager-resources/pull/129#issuecomment-640161770>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABMD3KMADNNBQBVVOXPYN33RVMTIRANCNFSM4NW2AJBQ>.

igor-makarov commented:
People frequently don't read the readme files very thoroughly 😂

I think a short, concise example would be great to put in the PR template - something along the lines of "e.g. `google.co.il` redirecting to `accounts.google.com` doesn't need an entry".

The goal of the PR template is to save time for the primary maintainers, at the expense of a minor inconvenience to individual contributors.

rmondello commented:
This is good. My attempt to synthesize everything said here into one concise line:

The new group has login pages on each of the included domains, and those login page accept accounts from the others. (If example1.website redirects to example2.website for login, there is no need to add an association.)

igor-makarov commented:
I would still insist on adding a familiar example, but I like the phrasing for the Readme.

rmondello commented:
The new group serves login pages on each of the included domains, and those login page accept accounts from the others. (For example, we don’t associate google.com.li to google.com, because google.com.li redirects to accounts.google.com for authentication.)

geoffcline commented:
ok i did the following things 

I revised the shared backend checkbox based on feedback from @rmondello and @igor-makarov. I added a section to the readme with a positive and negative example for shared backends. 

I also added another checklist section for change password URLs. 

looking forward to more feedback

rmondello commented on README.md:
```diff
@@ -36,6 +36,10 @@ An implementation of a parser for the Password Rules language that's written in
 
 The file [`quirks/websites-with-shared-credential-backends.json`](quirks/websites-with-shared-credential-backends.json) contains a list of groups of websites that are known to share the same credential backend. This can be used to offer contextually relevant accounts to users on website A, even if credentials were previously saved for website B.
```
I think we can achieve your goal here by amending this paragraph. Here’s my attempt, incorporating some of your great phrasing here.

> The file [`quirks/websites-with-shared-credential-backends.json`](quirks/websites-with-shared-credential-backends.json) contains a list of groups of websites that share the same credential backend and serve pages where users can sign in, accepting accounts from the others. For example, adding `first.website` and `second.website` means that `first.website` and `second.website` each serve a page (e.g. `first.website/login` and `second.website/login`) where the same accounts are valid for signing in, despite the different domains. It wouldn’t be appropriate to associate `google.com.li` to `google.com` because `google.com.li` redirects to `accounts.google.com` for sign-in, and `google.com.li` never serves a login page.
> 
> This data can be used by password managers to offer contextually relevant accounts to users on first.website, even if credentials were previously saved for second.website. This list should not be used as part of any user experience that releases...

geoffcline replied:
thanks! 🤝

igor-makarov commented on README.md:
```diff
@@ -36,6 +36,10 @@ An implementation of a parser for the Password Rules language that's written in
 
+In other words, a qualifying group serves login pages on each of the included domains. Those login pages accept accounts from the others. For example, `firstcompany.website/login` and `secondcompany.website/login` share the same backend (e.g., usernames, passwords, profiles), despite the different base domains.
+
+We don’t associate `google.com.li` to `google.com`, because `google.com.li` redirects to `accounts.google.com` for authentication.
```
typo: `.il` not `.li`

@geoffcline

geoffcline replied:
thanks!

geoffcline commented:
Thanks for your feedback @rmondello -- I revised the description in the readme

rmondello commented:
This is fantastic. Thank you!
